qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -826,6 +826,151 @@ def __new__(
         return instance
 
 
+def GaussianSquareDrag(
+    duration: Union[int, ParameterExpression],
+    amp: Union[float, ParameterExpression],
+    sigma: Union[float, ParameterExpression],
+    beta: Union[float, ParameterExpression],
+    width: Optional[Union[float, ParameterExpression]] = None,
+    angle: Optional[Union[float, ParameterExpression]] = 0.0,
+    risefall_sigma_ratio: Optional[Union[float, ParameterExpression]] = None,
+    name: Optional[str] = None,
+    limit_amplitude: Optional[bool] = None,
+) -> SymbolicPulse:
+    """A square pulse with a Drag shaped rise and fall
+
+    This pulse shape is similar to :class:`~.GaussianSquare` but uses
+    :class:`~.Drag` for its rise and fall instead of :class:`~.Gaussian`. The
+    addition of the DRAG component of the rise and fall is sometimes helpful in
+    suppressing the spectral content of the pulse at frequencies near the
+    fundamental frequency of the drive which can cause unwanted excitation of
+    spectators close in frequency to that fundamental frequency.
+
+    Exactly one of the ``risefall_sigma_ratio`` and ``width`` parameters has to be specified.
+
+    If ``risefall_sigma_ratio`` is not ``None`` and ``width`` is ``None``:
```

Comment:
Do we want to support both sigma and risefall ratio for new pulse? I heard experimentalists want to control the risefall ratio but we needed to keep width for GaussianSquare for backward compatibility. Since this is a new pulse, we don't have a backward compatibility issue. You added width because of the CmdDef parametric format?

Reply:
Yes, I made the CmdDef format match GaussianSquare, so it uses width.... We could still change it.

On the one hand, I think you are right that experimentalists prefer risefall because generally it is better to keep the shape of the edges of the pulse fixed and adjust the flat section. With `width`, that requires scanning `duration` and `width` together. With `risefall`, one can keep `risefall` fixed and just scan `duration`.  On the other hand, this pulse is a slight variation on GaussianSquare and there may be some value to making the two pulses behave as similarly as possible, so switching to GaussianSquareDrag just means changing the name and adding a `beta` value without worrying about changing  `width` to `risefall`. What do you think?

Reply:
Make sense. Let's keep width as long as it exists in the GaussianSquare.